src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorMqttMgr.java

```diff
@@ -0,0 +1,376 @@
+      mqttPublish(
+          CUMULOCITY_MQTT_TOPIC_SUS,
+          CConnectorApiMessageBuilder.configureHardware_110(serialNumber, modelNumber, revision),
+          MQTT_QOS_LEVEL,
+          MQTT_RETAIN);
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Unable to send hardware information to Cumulocity!");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Send basic firmware information
+    try {
+      SysControlBlock sysControlBlockInf = new SysControlBlock(SysControlBlock.INF);
+      String firmwareVersion = "\"" + sysControlBlockInf.getItem("CodeName") + "\"";
+      String firmwareDownloadUrlVersion = StringUtils.replace(firmwareVersion, ".", "_");
+      List serialNumberSplit = StringUtils.split(sysControlBlockInf.getItem("MbSerNum"), "-");
+      String firmwareDownloadPCode = serialNumberSplit.get(serialNumberSplit.size() - 1).toString();
+      String firmwareDownloadUrl =
+          "\""
+              + EWON_FIRMWARE_DOWNLOAD_WEBPAGE_URL
+              + "er"
+              + firmwareDownloadUrlVersion
+              + "p"
+              + firmwareDownloadPCode
+              + "_ma.edf"
+              + "\"";
+      mqttPublish(
+          CUMULOCITY_MQTT_TOPIC_SUS,
+          CConnectorApiMessageBuilder.setFirmware_115(
+              "Ewon Firmware", firmwareVersion, firmwareDownloadUrl),
+          MQTT_QOS_LEVEL,
+          MQTT_RETAIN);
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Unable to send firmware information to Cumulocity!");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Send basic software information
+    try {
+      // Create installed software array and add connector information as entry
+      InstalledSoftware[] installedSoftware = new InstalledSoftware[1];
+      installedSoftware[0] =
+          new InstalledSoftware(connectorName, connectorVersion, connectorDownloadUrl);
+      mqttPublish(
+          CUMULOCITY_MQTT_TOPIC_SUS,
+          CConnectorApiMessageBuilder.setSoftwareList_116(installedSoftware),
+          MQTT_QOS_LEVEL,
+          MQTT_RETAIN);
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Unable to send software information to Cumulocity!");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Send configuration to Cumulocity
+    sendConfigurationFileToCumulocity();
+
+    // Request pending operations (500)
+    try {
+      mqttPublish(
+          CUMULOCITY_MQTT_TOPIC_SUS,
+          CConnectorApiMessageBuilder.getPendingOperations_500(),
+          MQTT_QOS_LEVEL,
+          MQTT_RETAIN);
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Unable to request pending operations (Template 500) on MQTT connect!");
+      Logger.LOG_EXCEPTION(e);
+    }
+  }
+
+  /**
+   * Method for initializing connection
+   *
+   * @throws Exception if unable to initialize connection
+   */
+  public void initConnection() throws Exception {
+    Logger.LOG_DEBUG("Initializing MQTT connection...");
+
+    // Configure MQTT
+    setPortNonSecureDefault();
```

Comment:
again.. 1883. The IT team will not be happy about opening up outgoing to 1883. 

https://cumulocity.com/guides/device-sdk/mqtt/#connecting-via-mqtt

Reply:
See comment at https://github.com/hms-networks/flexy-cumulocity-connector/pull/2#discussion_r813173332 and the related comment above it